web/cypress/integration/Home/EditTest.spec.ts

```diff
@@ -1,43 +1,43 @@
-import {createTest, deleteTest, testId} from '../utils/Common';
+import {createTest, testRunPageRegex} from '../utils/Common';
 
 describe('Edit Test', () => {
-  before(() => {
-    createTest();
-  });
-
-  after(() => {
-    deleteTest();
-  });
-
-  beforeEach(() => {
-    cy.visit('http://localhost:3000/');
-  });
+  beforeEach(() => cy.visit('http://localhost:3000/'));
 
   it('should edit a test', () => {
-    cy.get(`[data-cy=test-actions-button-${testId}]`).click();
-    cy.get('[data-cy=test-card-edit]').click();
+    createTest(({testId}) => {
+      cy.visit(`http://localhost:3000?randomString=${23423489}`);
+
+      cy.get(`[data-cy=test-actions-button-${testId}]`).click();
+      cy.get('[data-cy=test-card-edit]').click();
 
-    cy.get('[data-cy=edit-test-form]').should('be.visible');
-    cy.get('[data-cy=create-test-name-input] input').clear().type('Edited Test');
+      cy.get('[data-cy=edit-test-form]').should('be.visible');
+      cy.get('[data-cy=create-test-name-input] input').clear().type('Edited Test');
 
-    cy.get('[data-cy=edit-test-submit]').click();
-    cy.get('[data-cy=test-details-name]').should('have.text', `Edited Test (v2)`);
+      cy.get('[data-cy=edit-test-submit]').click();
+      cy.get('[data-cy=test-details-name]').should('have.text', `Edited Test (v2)`);
 
-    cy.location('pathname').should('match', /\/test\/.*/i, {timeout: 10000});
+      cy.location('pathname').should('match', testRunPageRegex, {timeout: 20000});
+      // this does not work here
+      // deleteTest();
```

Comment:
any reason why it doesn't work? Shouldn't `deleteTest` receive an id instead of extracting it from the location?

Reply:
for some reason this deletion does not work. I have tried making deleteTest receive the testId.
but it fails the same way.